### PR TITLE
boards/common/weact-f4x1cx: create common WeAct boards

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

The common WeAct STM32F4x1 boards come in three variants.

 - WeAct-F411CE: 512k ROM / 128k RAM
 - WeAct-F401CE: 512k ROM / 96k RAM
 - WeAct-F401CC: 256k ROM / 64k RAM

On sites like [AliExpress](https://aliexpress.com/item/4000112571119.html) only the F401CC variant is advertised as the cheaper of two boards, ~~but it seems like you might randomly get a free update.~~
The [STM32F401CEU6 variant](https://de.aliexpress.com/item/4000346695587.html) is explicitly sold as such.

I previously ordered a WeAct-F401CC that actually had a STM32F401CC chip, but was broken, so I ordered another one…

The chips are pin compatible, so we can generalize the existing `weact-f411ce` board like it was done for the blue/blackpill boards.

### Testing procedure

I flashed a `weact-f411ce` and `weact-f401ce` via the USB bootloader and everything is working as before.

### CODE REVIEW

File: cpu/stm32/kconfig/f4/Kconfig.models [hunk 1]
```diff
@@ -6,6 +6,14 @@
 #
 
 # CPU models
+config CPU_MODEL_STM32F401CCU6
```
Comment (aabadie) [inline cpu/stm32/kconfig/f4/Kconfig.models:9]: There is no need to use that level of details in the cpu model name (`u6`). It just give information on the packing type.
Even ST doesn't give that level of details in [their online product list](https://www.st.com/en/microcontrollers-microprocessors/stm32f401.html#products).

Also have a look at #15129 where I already had to adapt the cpu model for the weact-f411ce.

File: boards/common/weact-f4x1cx/include/periph_conf.h [hunk 2]
```diff
@@ -36,7 +36,17 @@
 #define CLOCK_HSE               MHZ(25)
 
 #include "periph_cpu.h"
+
+#if defined(CPU_LINE_STM32F401xC) || defined(CPU_LINE_STM32F401xE)
+#define CONFIG_CLOCK_PLL_M      (25)
+#define CONFIG_CLOCK_PLL_N      (336)
```
Comment (aabadie) [inline boards/common/weact-f4x1cx/include/periph_conf.h:42]: You could move these particular defines directly in `f2f4f7/cfg_clock_default_84.h`, like it's done in `f2f4f7/cfg_clock_default_96.h`.
There's also no reason to have these defines non overridable: we should allow other possible configurable combinations for the PLL clock parameters.

Comment (benpicco) [reply boards/common/weact-f4x1cx/include/periph_conf.h]: Thank you for the hint, that makes it much cleaner.

File: cpu/stm32/kconfig/f4/Kconfig.models (hunk 1 above)
Comment (benpicco) [reply cpu/stm32/kconfig/f4/Kconfig.models:9]: I guess I'll just wait for  #15129  to be merged then any happily drop 108c0e57f50ae906f51fc4fd7a5d3f03cea94a10

File: boards/common/weact-f4x1cx/Kconfig [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2020 Inria
```
Comment (aabadie) [inline boards/common/weact-f4x1cx/Kconfig:1]: Inria ?

File: boards/weact-f401cc/Kconfig [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Copyright (c) 2020 Inria
```
Comment (aabadie) [inline boards/weact-f401cc/Kconfig:1]: Same here

File: boards/common/weact-f4x1cx/include/periph_conf.h [hunk 5]
```diff
@@ -36,7 +36,15 @@
...
-#include "clk_conf.h"
+
+#if defined(CPU_LINE_STM32F401xC) || defined(CPU_LINE_STM32F401xE)
+#include "f2f4f7/cfg_clock_default_84.h"
+#elif defined(CPU_LINE_STM32F411xE)
+#include "f2f4f7/cfg_clock_default_100.h"
+#else
+#error "Unknown CPU"
+#endif
+
```
Comment (aabadie) [inline boards/common/weact-f4x1cx/include/periph_conf.h:47]: #include "clk_conf.h"

File: cpu/stm32/include/clk/f2f4f7/cfg_clock_default_84.h [hunk 6]
```diff
@@ -36,11 +36,17 @@ extern "C" {
 /* The following parameters configure a 84MHz system clock with HSE (8MHz or
    16MHz) or HSI (16MHz) as PLL input clock */
```
Comment (aabadie) [inline cpu/stm32/include/clk/f2f4f7/cfg_clock_default_84.h:37]: This comment must be adapted as well

File: boards/weact-f401ce/doc.txt [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+@defgroup    boards_weact-f401ce WeAct-F401CE board
+@ingroup     boards
+@brief       Support for the WeAct-F401CE Board
+
+## Overview
+
+WeAct-F401CE is a board with the same form-factor as the blue/blackpill,
+but with an STM32F401CEU6 and a USB-C connector.
+
+It is available on sites like AliExpress for less than 3€.
```
Comment (aabadie) [inline boards/weact-f401ce/doc.txt:11]: Is this information important ? What about non EU members ?

Comment (benpicco) [reply boards/weact-f401ce/doc.txt:11]: I copied that from `blxxxpill` and adapted the price - can drop it.

Comment (aabadie) [reply boards/weact-f401ce/doc.txt:11]: I'm not insisting, it's just that RIOT users are not necessarily in EU :)

Comment (maribu) [reply boards/weact-f401ce/doc.txt:11]: AliExpress ships worldwide. And now that even James Bond villians prefer EUR over USD (see movie Quantum of Solace (2008)), we can expect everyone being able to make use of that info :-)

Comment (benpicco) [reply boards/weact-f401ce/doc.txt:11]: 🇪🇺🇪🇺🇪🇺

File: boards/common/weact-f4x1cx/include/board.h [hunk 8]
```diff
@@ -27,24 +27,16 @@ extern "C" {
 
 #include "mtd.h"
 #include "periph_cpu.h"
+#include "timex.h"
```
Comment (aabadie) [inline boards/common/weact-f4x1cx/include/board.h:30]: This include doesn't seem to be needed here.

File: boards/weact-f411ce/Makefile.include [hunk 9]
```diff
@@ -1,14 +1,5 @@
-INCLUDES += -I$(RIOTBOARD)/common/stm32/include
+# Product & Vendor ID taken from example firmware hat the board was shipped with.
```
Comment (aabadie) [inline boards/weact-f411ce/Makefile.include:1]:
```suggestion
# Product & Vendor ID taken from example firmware that the board was shipped with.
```
?

**Review by aabadie — COMMENTED** [review]
Minor comments, otherwise looks good.

**Comment by aabadie** [comment]
@kaspar030 do you have any idea why Murdock was triggered just now ? Only actions I made are 1. adding a review, 2. moving from "backlog" to "under review" in the sprint day project, 3. assign myself. Otherwise there were no new push, or label changed.

File: boards/common/weact-f4x1cx/include/board.h (hunk 8 above)
Comment (benpicco) [reply boards/common/weact-f4x1cx/include/board.h]: moved it to board.c, needed for `US_PER_MS` in `mtd_spi_nor_params_t`

**Review by aabadie — APPROVED** [review]
ACK, please squash!

**Pull request merged**